parsers/ES.py

```diff
@@ -37,45 +21,107 @@
 from .lib.config import refetch_frequency
 from .lib.exceptions import ParserException
 
-SOURCE = "demanda.ree.es"
+#    Zone name Cheat Sheet
+
+#    ZoneKey("ES"): "IberianPeninsula"
+#    ZoneKey("ES-CE"): "Ceuta"
+#    ZoneKey("ES-CN-FVLZ"): "LanzaroteFuerteventura"
+#    ZoneKey("ES-CN-GC"): "GranCanaria"
+#    ZoneKey("ES-CN-HI"): "ElHierro"
+#    ZoneKey("ES-CN-IG"): "Gomera"
+#    ZoneKey("ES-CN-LP"): "LaPalma"
+#    ZoneKey("ES-CN-TE"): "Tenerife"
+#    ZoneKey("ES-IB-FO"): "Formentera"
+#    ZoneKey("ES-IB-IZ"): "Ibiza"
+#    ZoneKey("ES-IB-MA"): "Mallorca"
+#    ZoneKey("ES-IB-ME"): "Menorca"
+#    ZoneKey("ES-ML"): "Melilla"
+
+
+DATA_MAPPING = {
+    "dem": "demand",
+    "nuc": "production.nuclear",
+    "die": "production.oil",
+    "genAux": "production.oil",
+    "gas": "production.gas",
+    "gf": "production.gas",
+    "eol": "production.wind",
+    "cc": "production.gas",
+    "vap": "production.oil",
+    "fot": "production.solar",
+    "sol": "production.solar",
+    "hid": "storage.hydro",
+    "car": "production.coal",
+    "resid": "production.biomass",
+    "termRenov": "production.unknown",
+    "cogenResto": "production.unknown",
+    "tnr": "production.unknown",
+    "trn": "production.unknown",
+    "otrRen": "production.unknown",
+    "cogen": "production.gas",
+    "cb": "exchange.pe_ma",
+    "icb": "exchange.pe_ma",
+    "emm": "exchange.ma_me",
+    "emi": "exchange.ma_ib",
+    "eif": "exchange.ib_fo",
+    "inter": "exchange.int",
+}
```

Comment:
Why prefix these are require us to split the strings later on instead of having multiple lists which is both faster and easier to maintain?